lib/server.go

```diff
@@ -30,8 +38,31 @@ func (s *Server) Dial(url string) error {
 }
 
 // Handler returns an HTTP handler that can be directly connected to the router.
-func (s *Server) Handler() http.Handler {
-	return websocket.Handler(s.handle)
+func (s *Server) Handler() func(http.ResponseWriter, *http.Request) {
+	if s.Upgrader == nil {
+		s.Upgrader = DefaultUpgrader
+	}
+	return func(w http.ResponseWriter, r *http.Request) {
+		ws, err := s.Upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			s.catchError(err)
+			return
+		}
+		channel, err := s.Connection.Channel()
+		if err != nil {
+			s.catchError(err)
+			return
+		}
+		requestHandler := &handler{
```

Comment:
I'd suggest creating a constructor function for handler. It should make your function more readable.